### DISCUSSION TITLE
Useless multiplication?

### DISCUSSION THREAD

**Discussion by ghley** [DanielChappuis/reactphysics3d#277] [General]
I might be wrong here as I haven't worked through all the code so far and I'm assuming inverseMass is always 1/m but isn't in DynamicsSystem.cpp line 152

`                mRigidBodyComponents.mConstrainedLinearVelocities[i] = mRigidBodyComponents.mConstrainedLinearVelocities[i] + timeStep *
                                                                       mRigidBodyComponents.mInverseMasses[i] * mRigidBodyComponents.mLinearLockAxisFactors[i] *
                                                                       mRigidBodyComponents.mMasses[i] * mGravity;`

the inversMass * Mass always 1 and thus irrelevant?

**Comment by DanielChappuis** [comment]
Hello,

Yes, it seems to be useless but the only thing is that  the 'inverseMass' of a body can also be zero that is useful to simulate a static body that cannot move. Therefore, if we remove the `inverseMass * mass` multiplication we cannot simulate a static body (that has zero inverse mass).

**Reply by ghley** [reply]
That makes sense, so instead of checking for the body to be static it's all just integrated in the calculation.

Thanks!

**Reply by DanielChappuis** [reply]
Yes exactly. Note that it is also possible for a user to set the inverse mass of a body to be zero even for a body that is not static.
